src/tree/LocalRootTreeItemBase.ts

```diff
@@ -112,11 +124,18 @@ export abstract class LocalRootTreeItemBase<TItem extends ILocalItem, TProperty
             this._currentItems = this._itemsFromPolling || await this.getSortedItems();
             this._itemsFromPolling = undefined;
             this.failedToConnect = false;
+            this._currentDockerStatus = 'Running';
         } catch (error) {
             this._currentItems = undefined;
             this.failedToConnect = true;
             context.telemetry.properties.failedToConnect = 'true';
-            return this.getDockerErrorTreeItems(context, error);
+
+            if (!this._currentDockerStatus) {
```

Comment:
I would not wait inside `loadMoreChildrenImpl` to do a Docker status refresh, even if this is just in the error code path. Can we do the status refresh asynchronously?

Reply:
The status is needed to construct the error nodes. EIther 'docker installed?' or 'docker running?'. This is the same behavior as how the code waits on getSortedItems() in line 124.

Reply:
No it isn't quite the same--you are introducing an extra wait to spawn a process and do `docker -v` if initial tree population attempt fails.

Assuming Docker is **not installed**, but the user has Docker view open--how often will we run `docker -v` after this change?

Reply:
This gets called only when there is a change in dockerinstall status which is updated by hasChanged() function. For user with no docker installed, this will gets called once per pane until the docker status change. hasChanged() gets called for every 2 seconds.

Reply:
Also note that line 133 if condition check will make sure that it gets called only when the `this._currentDockerStatus` is undefined which is the first time when user opens the docker view. The next refresh will not await.

Reply:
Just realized this code path will be executed at most once during a session? If that is indeed the case, i am much less worried. Feel free to resolve if my assumption is right

Reply:
That is right. This will gets called once per session.